### PR TITLE
Build static Index website from survey site admin 

### LINKED ISSUE #856: Expose Index generation actions to site admin dashboard
Index site generation (#853) and export to S3 (#854) should be available as UI actions for site admins using the admin dashboard.

Note, export to S3 may be part of the same action as site generation.

### PR DESCRIPTION
**Pull request by brew**
This PR adds an admin controller action to build and deploy a static Index website to an AWS S3 bucket.

The action uses the existing `generate_index` script with appropriate arguments for the domain and year.

Closes #856.

### CODE REVIEW

File: census/controllers/admin.js [hunk 1]
```diff
@@ -66,10 +69,31 @@ let loadQuestionSets = function(req, res) {
...
+  /*
+  Generate and deploy an Index site for the site specified in the request.
+  */
+
+  // Set env for process to include DEBUG option, for more comprehensive
+  // logging.
+  const env = process.env;
+  // env.DEBUG = 'metalsmith-godi-*';
+
+  exec(`node index/generate.js ${res.locals.domain} -dsy ${res.locals.surveyYear}`,
```
Comment (roll) [inline census/controllers/admin.js]: `res.locals` has been set somewhere on the server so it's trusted data to inject into exec?

Comment (brew) [inline census/controllers/admin.js]: Thanks - I'll chase this through and see if `domain` can be set externally and misused.

Comment (brew) [inline census/controllers/admin.js]: In theory, the value for `res.locals.domain` is checked before it reaches this action, but it wouldn't take many code changes to open this up to abuse.

Using `exec` with user-provided arguments does have the possibility of misuse. I've committed a change to use `execFile` instead which doesn't spawn a new shell, but instead calls the passed file directly.

Comment (roll) [inline census/controllers/admin.js]: :+1:

**Pull request merged**